scripts/grt/export.py

```diff
@@ -277,6 +278,47 @@ def annotate(label, human_label=None):
     handle_job.close()
     annotate('export_jobs_end')
 
+    annotate('export_datasets_start', 'Exporting Datasets')
+    handle_datasets = open(REPORT_BASE + '.datasets.tsv', 'w')
+    handle_datasets.write('\t'.join(('job_id', 'dataset_id', 'extension', 'file_size', 'name', 'param_name')) + '\n')
+    for offset_start in range(last_job_sent, end_job_id, args.batch_size):
+        logging.debug("Processing %s:%s", offset_start, min(end_job_id, offset_start + args.batch_size))
+        for job in sa_session.query(model.Job) \
+                .filter(model.Job.id > offset_start) \
+                .filter(model.Job.id <= min(end_job_id, offset_start + args.batch_size)) \
+                .all():
+
+            # If the tool is blacklisted, exclude everywhere
+            if job.tool_id in blacklisted_tools:
+                continue
+
+            input_datasets=(job.input_datasets)
+            for input_dataset in input_datasets:
+                hda = input_dataset.dataset
+                dataset = hda.dataset
+                datasets=({'dataset_id': str(hda.dataset_id),
```

Comment:
This probably works, but given we could be processing millions of datasets we might want to consider leveraging the fact that Galaxy has a `job_to_input_dataset` table and `dataset` table has `total_size` attribute. This should allow us to get all the data directly using the initial query without accessing object properties here and below (which often fire of their own queries).

Reply:
:+1:

Reply:
I looked at these two tables: JobToInputDatasetAssociation and Dataset. They have almost everything we need except for an extension attribute. I thought it would be in the Dataset table... Any idea where to get the file extension?

Reply:
it should be the `ext` property on the hda

Reply:
the column is actually named `extension` in the history_dataset_association table, ext is a synonym used in the Model